### PR TITLE
Lift

### PR DESCRIPTION
**Pull request by kcjesse02**
Pull Request Checklist:
- [x] You are able to understand what the code is doing on a basic level with the use of comments
- [x] Code is formatted in a way that isn't confusing
- [x] builds cleanly
- [x] builds *to the robot* cleanly
- [x] Code has been tested on the robot/roadkill/other mechanical setup and completes the task it needs to complete

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 1]
```diff
@@ -7,104 +7,118 @@
...
+    liftMotor.configFactoryDefault();
+    liftMotor.enableCurrentLimit(true);
+    liftMotor.configPeakCurrentLimit(35);
+    liftMotor.configContinuousCurrentLimit(20);
+    liftMotor.configPeakCurrentDuration(250);
+    liftMotor.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+    liftMotor.configPeakOutputForward(1.0);
+    
+    // TODO: add encoder limits
+    max_encoder = ___;
```
Comment (r-wallis) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: doesn't compile for me (max_encoder = __, min_encoder = __)

**Review by classygirl44 — DISMISSED** [review]

**Review by Linkd-Aether — DISMISSED** [review]

File: src/main/java/frc/robot/commands/LiftControls.java [hunk 2]
```diff
@@ -7,19 +7,25 @@
 
 package frc.robot.commands;
 
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.OI;
+import frc.robot.subsystems.instances.Lift;
```
Comment (jjesse1972) [inline src/main/java/frc/robot/commands/LiftControls.java]: we shouldnt need Lift. as a matter of fact, we want to make sure we dont. Access all lift capabilities through LiftInterface only

File: src/main/java/frc/robot/commands/LiftControls.java [hunk 3]
```diff
@@ -7,19 +7,25 @@
...
+  WinchInterface winch;
 
   /**
    * Creates a new LiftControls.
    */
-  public LiftControls(LiftInterface lift_) {
+  public LiftControls(LiftInterface lift_, WinchInterface winch_) {
     lift = lift_;
+    winch = winch_;
     addRequirements((SubsystemBase)lift);
```
Comment (jjesse1972) [inline src/main/java/frc/robot/commands/LiftControls.java]: We need an 'addRequirements' for the drivetrain (winch) subsystem.

File: src/main/java/frc/robot/commands/LiftControls.java [hunk 4]
```diff
@@ -31,6 +37,14 @@ public void initialize() {
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
+    SmartDashboard.putNumber("Encoder Ticks", lift.getEncoderTicks());
+
```
Comment (jjesse1972) [inline src/main/java/frc/robot/commands/LiftControls.java:41]: comment the reasoning here

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 5]
```diff
@@ -7,104 +7,122 @@
...
+  private static int min_encoder;
+  private static int max_encoder;
+  private int encoder_ticks;
+  
+
   /**
    * Creates a new Lift.
    */
-  Solenoid solenoid = new Solenoid(7);
+  // Solenoid solenoid = new Solenoid(7);
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: remove this line

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 6]
```diff
@@ -7,104 +7,122 @@
...
   
-  // that sets minimum possible value the potentiometer  can return (when the voltage from the potentiometer is 0 (the minimum))
-  private final double minLiftExtension = 0.0;
-  // that sets maximum possible value the potentiometer can return (when the voltage from the potentiometer is 1 (the maximum))
-  private final double maxLiftExtension = 1.0;
+  // // that sets minimum possible value the potentiometer  can return (when the voltage from the potentiometer is 0 (the minimum))
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: remove these commented out lines

**Review by r-wallis — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 7]
```diff
@@ -7,104 +7,122 @@
...
+    
+    // // TODO: add encoder limits
+    // max_encoder = ___;
+    // min_encoder = 0;
 
-  public Lift() {
 
     // Enables 2-bit averaging
-    potentiometerValue.setAverageBits(2);
+    // potentiometerValue.setAverageBits(2);
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 8]
```diff
@@ -7,104 +7,122 @@
...
-    liftExtension();
-    isPinned();
-
+    encoder_ticks = liftMotor.getSelectedSensorPosition();
   }
 
-  public void setBrakeOn() {
-    // Turns on the brake
+  public void liftExtend(double power) {
+    // if (liftMotor.getSelectedSensorPosition() > min_encoder && liftMotor.getSelectedSensorPosition() < max_encoder){
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java:55]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 9]
```diff
@@ -7,104 +7,122 @@
...
-  }
+  // public void setBrakeOn() {
+  //   // Turns on the brake
+  // }
 
-  public boolean isBrakeSet() {
-    // Checks to see if the brake is set
-    return true;
-  }
+  // public void setBrakeOff() {
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 10]
```diff
@@ -7,104 +7,122 @@
...
+  //   // Turns off the brake
+  // }
 
-  public double liftExtension() {
-
-    // returns in the scale from minLiftExtention to maxLiftExtention
-    return potentiometer.get();
+  // public boolean isBrakeSet() {
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/instances/Lift.java [hunk 11]
```diff
@@ -7,104 +7,122 @@
...
   }
 
-  public boolean isLiftFullyExtended() {
-    // Checks to see if the lift is fully extended
-    if (liftExtension() >= physicalMaxExtension) {
-      return true;
-    } else {
-      return false;
-    }
+  // public boolean isLiftFullyRetracted() {
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/instances/Lift.java:69]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/interfaces/LiftInterface.java [hunk 12]
```diff
@@ -10,17 +10,16 @@
 public interface LiftInterface {
 
     // brake control
-    void setBrakeOn();
-    void setBrakeOff();
-    boolean isBrakeSet();
+    // void setBrakeOn();
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/interfaces/LiftInterface.java]: remove these commented lines unless we intend to uncomment soon.

File: src/main/java/frc/robot/subsystems/interfaces/LiftInterface.java [hunk 13]
```diff
@@ -10,17 +10,16 @@
...
+    // void setBrakeOff();
+    // boolean isBrakeSet();
 
     // potentiometer
-    boolean isLiftFullyExtended();
-    boolean isLiftFullyRetracted();
+    // boolean isLiftFullyExtended();
```
Comment (jjesse1972) [inline src/main/java/frc/robot/subsystems/interfaces/LiftInterface.java:12]: remove these commented lines unless we intend to uncomment soon.

**Review by jjesse1972 — CHANGES_REQUESTED** [review]

**Review by jjesse1972 — APPROVED** [review]

**Review by classygirl44 — APPROVED** [review]

**Pull request merged**